feat: 그룹 생성 및 카테고리 목록 조회

ownue opened the pull request:
## 📌 관련 이슈
#9

## ✨ 작업 내용
- 그룹을 생성하는 API 추가
- 프론트를 위해 전체 카테고리 목록 조회 API 추가
- ErrorCode 활용해 예외 처리 통일

## 📸 스크린샷(선택)
X

## 📚 레퍼런스 (또는 새로 알게 된 내용) 혹은 궁금한 사항들
X

ownue committed:
#9 feat: 그룹 생성 및 카테고리 목록 조회

- 그룹을 생성하는 API 추가
- 그룹 생성 시, isPublic이 JPA 단에서 오류가 나 변수명을 opened로 변경 (속성명은 유지)
- 프론트를 위해 전체 카테고리 목록 조회 API 추가

hyomee2 commented on src/main/java/com/example/muaring/domain/group/entity/Group.java:
```diff
@@ -40,8 +40,18 @@ public class Group extends BaseEntity {
 
     @Column(name = "is_public", nullable = false, columnDefinition = "BOOLEAN DEFAULT TRUE")
-    private Boolean isPublic = true;
+    private Boolean opened = true;  // isPublic -> opened 로 변경
```
혹쉬.. 이거는 왜 변경된걸까요?!??!

ownue replied:
자꾸 오류나서 바꿔버렸는데 제 오탈자 문제였나바요,, 다시 수정하러 갑니다 슝슝;;

hyomee2 replied:
사실 크게 상관은 없는데.. 궁금해서 물어봤어요!!!

ownue replied:
빠르게 수정했습니다 ✌️

hyomee2 commented on src/main/java/com/example/muaring/domain/group/service/GroupCategoryService.java:
```diff
@@ -3,19 +3,17 @@
-        this.groupCategoryRepository = groupCategoryRepository;
-    }
-
     public List<GroupCategoryResponseDto> findAllCategories() {
```
굳입니다

hyomee2 commented on src/main/java/com/example/muaring/domain/group/service/GroupService.java:
```diff
@@ -0,0 +1,65 @@
+    // 그룹 생성
+    public GroupCreateResponseDto createGroup(GroupCreateRequestDto requestDto) {
+        Member admin = memberRepository.findById(requestDto.getAdminId())
+                .orElseThrow(() -> new EntityNotFoundException("사용자를 찾을 수 없습니다."));
```
이거! 혹시 ErrorCode 만들어서 사용해주실 수 있을까요? Test 쪽에 보면 나와있을텐데, 도메인exception 만들고 errorcode에 enum으로 추가해서 해주시면 될 것 같아요!!

ownue replied:
넵!!

hyomee2 commented on src/main/java/com/example/muaring/domain/group/service/GroupService.java:
```diff
@@ -0,0 +1,65 @@
+
+        for (Long categoryId : groupCategoryIds) {
+            GroupCategory groupCategory = groupCategoryRepository.findById(categoryId)
+                    .orElseThrow(() -> new EntityNotFoundException("Category not found with id: " + categoryId));
```
여기도 위에서 말한 errorcode로 통일하면 좋을거 같아요!

ownue committed:
#9 refactor: 예외 처리 로직 개선 및 ErrorCode 도입

- 기존의 ErrorCode(ENUM)을 interface를 상속받는 구조로 변경
- Member 도메인과 Group 도메인에 ErrorCode 및 Exception 추가하여 예외 처리 통일

ownue committed:
#9 refactor: 그룹 카테고리 seed 로직 리팩토링

- findAll()을 통해 DB 접근 횟수 감축

romdyfo commented on src/main/java/com/example/muaring/domain/group/service/GroupService.java:
```diff
@@ -0,0 +1,65 @@
+    private final MemberRepository memberRepository;
+
+    // 그룹 생성
+    public GroupCreateResponseDto createGroup(GroupCreateRequestDto requestDto) {
```
여기서 adminId를 클라이언트가 전달하게 하기 보다는 나중에 로그인 기능이 완료되면 SecurityContext에서 현재 사용자 ID를 추출하는 방식으로 하면 좋을 것 같아요!

ownue replied:
API 문서에 적어두긴 했는데 아직 회원 관련 기능 개발이 안 돼서 테스트 용도로 이렇게 해둔 거고 해당 기능 개발된 이후에 수정할 예정입니다!! 😁😁 감사합니다!

romdyfo replied:
앗 네 감사합니다!!

romdyfo commented on src/main/java/com/example/muaring/domain/group/service/GroupService.java:
```diff
@@ -0,0 +1,65 @@
+    private final MemberRepository memberRepository;
+
+    // 그룹 생성
+    public GroupCreateResponseDto createGroup(GroupCreateRequestDto requestDto) {
```
crateGroup 메서드 위에 @Transactional 붙이면 좋을 것 같아요!!

ownue replied:
해당 부분 수정해서 다시 커밋했습니다!

romdyfo replied:
네 감사합니다!!☺️

ownue committed:
chore: 어노테이션 추가

- GroupService의 createGroup 메서드 위에 @Transactional 어노테이션을 추가

hyomee2 reviewed (approved):
고생하셨습니당!
